### PR TITLE
Move Aruco tutorials and samples to main repo

### PR DESCRIPTION
**Pull request by stopmosk**
merge with https://github.com/opencv/opencv_contrib/pull/3401
merge with https://github.com/opencv/opencv_extra/pull/1143

### Pull Request Readiness Checklist

See details at https://github.com/opencv/opencv/wiki/How_to_contribute#making-a-good-pull-request

- [x] I agree to contribute to the project under Apache 2 License.
- [x] To the best of my knowledge, the proposed patch is not based on a code under GPL or another license that is incompatible with OpenCV
- [x] The PR is proposed to the proper branch
- [ ] There is a reference to the original bug report and related work
- [ ] There is accuracy test, performance test and test data in opencv_extra repository, if applicable
      Patch to opencv_extra has the same branch name.
- [ ] The feature is well documented and sample code can be built with the project CMake

```
**WIP**
force_builders=linux,docs
build_contrib:Docs=OFF
```

### CODE REVIEW

File: modules/objdetect/samples/detect_markers.cpp [hunk 1]
```diff
@@ -0,0 +1,208 @@
+/*
+By downloading, copying, installing or using the software you agree to this
```
Comment (alalek) [inline modules/objdetect/samples/detect_markers.cpp:2]: no license in samples

File: doc/tutorials/objdetect/aruco_board_detection/aruco_board_detection.markdown [hunk 2]
```diff
@@ -0,0 +1,298 @@
...
+
+@prev_tutorial{tutorial_aruco_detection}
+@next_tutorial{tutorial_charuco_detection}
+
+An ArUco Board is a set of markers that acts like a single marker in the sense that it provides a
+single pose for the camera.
+
+The most popular board is the one with all the markers in the same plane, since it can be easily printed:
+
+![](images/gboriginal.png)
```
Comment (alalek) [inline doc/tutorials/objdetect/aruco_board_detection/aruco_board_detection.markdown:12]: Real life photos must be in JPEG format.
PNG blows up size in these cases.

File: doc/tutorials/objdetect/aruco_board_detection/aruco_board_detection.markdown [hunk 3]
```diff
@@ -0,0 +1,298 @@
...
+if(markerIds.size() > 0) {
+    // Get object and image points for the solvePnP function
+    cv::Mat objPoints, imgPoints;
+    board->matchImagePoints(markerCorners, markerIds, objPoints, imgPoints);
+
+    // Find pose
+    cv::solvePnP(objPoints, imgPoints, cameraMatrix, distCoeffs, rvec, tvec);
+}
+
+@endcode
```
Comment (alalek) [inline doc/tutorials/objdetect/aruco_board_detection/aruco_board_detection.markdown:72]: Snippets should be used instead of manual **non-verified** code.

Comment (stopmosk) [reply doc/tutorials/objdetect/aruco_board_detection/aruco_board_detection.markdown:72]: What should I do if the example contains a lot of extra code that I would **not** like to include in the tutorial?

Comment (alalek) [reply doc/tutorials/objdetect/aruco_board_detection/aruco_board_detection.markdown:72]: Take a look on `cv::reduce` on docs.opencv.org as example.
You could define any part of the code to include into documentation.

File: modules/objdetect/samples/detect_markers.cpp (hunk 1 above)
Comment (AleksandrPanov) [reply modules/objdetect/samples/detect_markers.cpp]: fixed

File: doc/tutorials/objdetect/aruco_board_detection/aruco_board_detection.markdown (hunk 2 above)
Comment (AleksandrPanov) [reply doc/tutorials/objdetect/aruco_board_detection/aruco_board_detection.markdown]: fixed

File: doc/tutorials/objdetect/aruco_board_detection/aruco_board_detection.markdown (hunk 3 above)
Comment (AleksandrPanov) [reply doc/tutorials/objdetect/aruco_board_detection/aruco_board_detection.markdown]: replaced by snippets

File: doc/tutorials/objdetect/aruco_board_detection/aruco_board_detection.markdown [hunk 4]
```diff
@@ -0,0 +1,194 @@
...
+In this case the margin is 10.
+- Finally, the size of the marker border, similarly to ```generateImageMarker()``` function. The default value is 1.
+
+The output image will be something like this:
+
+![](images/board.png)
+
+A full working example of board creation is included in the `samples/cpp/tutorial_code/objectDetection/create_board.cpp`.
+
+Note: The samples now take input via commandline via the [OpenCV Commandline Parser](http://docs.opencv.org/trunk/d0/d2e/classcv_1_1CommandLineParser.html#gsc.tab=0). For this file the example parameters will look like
```
Comment (asmorkalov) [inline doc/tutorials/objdetect/aruco_board_detection/aruco_board_detection.markdown:124]: Please use doxyen/markdown anchor, but not http links.

File: doc/tutorials/objdetect/aruco_board_detection/aruco_board_detection.markdown [hunk 5]
```diff
@@ -0,0 +1,194 @@
...
+
+Sample video:
+
+@htmlonly
+<iframe width="420" height="315" src="https://www.youtube.com/embed/Q1HlJEjW_j0" frameborder="0" allowfullscreen></iframe>
+@endhtmlonly
+
+A full working example is included in the `detect_board.cpp` inside the `samples/cpp/tutorial_code/objectDetection/`.
+
+Note: The samples now take input via commandline via the [OpenCV Commandline Parser](http://docs.opencv.org/trunk/d0/d2e/classcv_1_1CommandLineParser.html#gsc.tab=0). For this file the example parameters will look like
```
Comment (asmorkalov) [inline doc/tutorials/objdetect/aruco_board_detection/aruco_board_detection.markdown:137]: The same for links.

File: doc/tutorials/objdetect/aruco_board_detection/aruco_board_detection.markdown [hunk 6]
```diff
@@ -0,0 +1,194 @@
...
+
+The most popular board is the one with all the markers in the same plane, since it can be easily printed:
+
+![](images/gboriginal.jpg)
+
+However, boards are not limited to this arrangement and can represent any 2d or 3d layout.
+
+The difference between a Board and a set of independent markers is that the relative position between
+the markers in the Board is known a priori. This allows that the corners of all the markers can be used for
+estimating the pose of the camera respect to the whole Board.
```
Comment (asmorkalov) [inline doc/tutorials/objdetect/aruco_board_detection/aruco_board_detection.markdown:18]: board, capital letter is not required.

File: doc/tutorials/objdetect/aruco_board_detection/aruco_board_detection.markdown [hunk 7]
```diff
@@ -0,0 +1,194 @@
...
+@endhtmlonly
+
+A full working example is included in the `detect_board.cpp` inside the `samples/cpp/tutorial_code/objectDetection/`.
+
+Note: The samples now take input via commandline via the [OpenCV Commandline Parser](http://docs.opencv.org/trunk/d0/d2e/classcv_1_1CommandLineParser.html#gsc.tab=0). For this file the example parameters will look like
+@code{.cpp}
+    -w=5 -h=7 -l=100 -s=10
+    -v=/path_to_aruco_tutorials/aruco_board_detection/images/gboriginal.jpg
+    -c=/path_to_aruco_samples/tutorial_camera_params.yml
+    -cd=/path_to_aruco_samples/tutorial_dict.yml
```
Comment (asmorkalov) [inline doc/tutorials/objdetect/aruco_board_detection/aruco_board_detection.markdown:142]: it should be relative to opencv repo root.

**Comment by asmorkalov** [comment]
General comment:
`refineDetectedMarkers`, `detectMarkers` and others are class methods, but not free functions. Need to fix wording.

File: samples/cpp/tutorial_code/objectDetection/aruco_samples_utility.hpp [hunk 8]
```diff
@@ -0,0 +1,48 @@
+#include <opencv2/highgui.hpp>
+#include <opencv2/objdetect/aruco_detector.hpp>
+#include <opencv2/calib3d.hpp>
+#include <ctime>
+
+namespace {
+inline static bool readCameraParameters(std::string filename, cv::Mat &camMatrix, cv::Mat &distCoeffs) {
```
Comment (asmorkalov) [inline samples/cpp/tutorial_code/objectDetection/aruco_samples_utility.hpp:7]: `const std::string&`

File: samples/cpp/tutorial_code/objectDetection/create_marker.cpp [hunk 9]
```diff
@@ -0,0 +1,77 @@
...
+    bool showImage = parser.get<bool>("si");
+
+    String out = parser.get<String>(0);
+
+    if(!parser.check()) {
+        parser.printErrors();
+        return 0;
+    }
+
+    aruco::Dictionary dictionary = aruco::getPredefinedDictionary(0);
```
Comment (asmorkalov) [inline samples/cpp/tutorial_code/objectDetection/create_marker.cpp:48]: please use named constant instead 0.

File: samples/cpp/tutorial_code/objectDetection/detect_markers.cpp [hunk 10]
```diff
@@ -0,0 +1,170 @@
...
+    if(parser.has("v")) {
+        video = parser.get<String>("v");
+    }
+
+    if(!parser.check()) {
+        parser.printErrors();
+        return 0;
+    }
+
+    aruco::Dictionary dictionary = aruco::getPredefinedDictionary(0);
```
Comment (asmorkalov) [inline samples/cpp/tutorial_code/objectDetection/detect_markers.cpp:72]: Please use named constant for dictionary.

**Comment by asmorkalov** [comment]
I propose to replace png images with jpg to reduce patch size.

File: doc/tutorials/objdetect/aruco_board_detection/aruco_board_detection.markdown [hunk 11]
```diff
@@ -0,0 +1,298 @@
...
+- The obtained pose is usually more accurate since a higher amount of point correspondences (marker
+corners) are employed.
+
+Board Detection
+-----
+
+A Board detection is similar to the standard marker detection. The only difference is in the pose estimation step.
+In fact, to use marker boards, a standard marker detection should be done before estimating the Board pose.
+
+To perform pose estimation for boards, you should use ```#cv::solvePnP``` function, as shown below:
```
[the remark this answers is not in the document]
Comment (AleksandrPanov) [reply doc/tutorials/objdetect/aruco_board_detection/aruco_board_detection.markdown]: need to use:
solvePnP() or \`solvePnP()\`
I suggest to use \`solvePnP()\` to improve `.markdown` readability:
image

File: doc/tutorials/objdetect/aruco_board_detection/aruco_board_detection.markdown [hunk 12]
```diff
@@ -0,0 +1,298 @@
...
+
+The parameters are:
+
+- ```objPoints```, ```imgPoints```: object and image points, matched with ```matchImagePoints```, which, in turn, takes as input ```markerCorners``` and ```markerIds```: structures of detected markers from ```detectMarkers()``` function).
+- ```board```: the ```Board``` object that defines the board layout and its ids
+- ```cameraMatrix``` and ```distCoeffs```: camera calibration parameters necessary for pose estimation.
+- ```rvec``` and ```tvec```: estimated pose of the Board. If not empty then treated as initial guess.
+- The function returns the total number of markers employed for estimating the board pose.
+
+The ```drawFrameAxes()``` function can be used to check the obtained pose. For instance:
```
[the remark this answers is not in the document]
Comment (AleksandrPanov) [reply doc/tutorials/objdetect/aruco_board_detection/aruco_board_detection.markdown]: drawFrameAxes() or \`drawFrameAxes()\`

File: doc/tutorials/objdetect/aruco_board_detection/aruco_board_detection.markdown [hunk 13]
```diff
@@ -0,0 +1,194 @@
...
+
+![Board with occlusions](images/gbocclusion.jpg)
+@note The center and direction of the axes has been changed
+
+As it can be observed, although some markers have not been detected, the Board pose can still be estimated from the rest of markers.
+
+Grid Board
+-----
+
+Creating the ```cv::aruco::Board``` object requires specifying the corner positions for each marker in the environment.
```
[the remark this answers is not in the document]
Comment (AleksandrPanov) [reply doc/tutorials/objdetect/aruco_board_detection/aruco_board_detection.markdown]: added \`cv::aruco::Board\`

File: doc/tutorials/objdetect/aruco_board_detection/aruco_board_detection.markdown (hunk 4 above)
Comment (AleksandrPanov) [reply doc/tutorials/objdetect/aruco_board_detection/aruco_board_detection.markdown]: fixed:
image

File: doc/tutorials/objdetect/aruco_board_detection/aruco_board_detection.markdown (hunk 5 above)
Comment (AleksandrPanov) [reply doc/tutorials/objdetect/aruco_board_detection/aruco_board_detection.markdown]: fixed

**Comment by AleksandrPanov** [comment]
> General comment: `refineDetectedMarkers`, `detectMarkers` and others are class methods, but not free functions. Need to fix wording.

fixed

File: samples/cpp/tutorial_code/objectDetection/aruco_samples_utility.hpp (hunk 8 above)
Comment (AleksandrPanov) [reply samples/cpp/tutorial_code/objectDetection/aruco_samples_utility.hpp]: fixed

File: doc/tutorials/objdetect/aruco_board_detection/aruco_board_detection.markdown (hunk 7 above)
Comment (AleksandrPanov) [reply doc/tutorials/objdetect/aruco_board_detection/aruco_board_detection.markdown]: fixed

File: doc/tutorials/objdetect/aruco_board_detection/aruco_board_detection.markdown (hunk 6 above)
Comment (AleksandrPanov) [reply doc/tutorials/objdetect/aruco_board_detection/aruco_board_detection.markdown]: fixed

File: samples/cpp/tutorial_code/objectDetection/create_marker.cpp (hunk 9 above)
Comment (AleksandrPanov) [reply samples/cpp/tutorial_code/objectDetection/create_marker.cpp]: fixed

File: samples/cpp/tutorial_code/objectDetection/detect_markers.cpp (hunk 10 above)
Comment (AleksandrPanov) [reply samples/cpp/tutorial_code/objectDetection/detect_markers.cpp]: fixed

**Comment by AleksandrPanov** [comment]
> I propose to replace png images with jpg to reduce patch size.

The remaining PNG images are smaller than the JPG images.

File: doc/tutorials/objdetect/aruco_board_detection/aruco_board_detection.markdown (hunk 11 above)
[the remark this answers is not in the document]
Comment (AleksandrPanov) [reply doc/tutorials/objdetect/aruco_board_detection/aruco_board_detection.markdown]: image

File: doc/tutorials/objdetect/aruco_board_detection/aruco_board_detection.markdown [hunk 14]
```diff
@@ -0,0 +1,194 @@
...
+- The function returns the total number of markers employed for estimating the board pose.
+
+The drawFrameAxes() function can be used to check the obtained pose. For instance:
+
+![Board with axis](images/gbmarkersaxis.jpg)
+
+And this is another example with the board partially occluded:
+
+![Board with occlusions](images/gbocclusion.jpg)
+@note The center and direction of the axes has been changed
```
Comment (asmorkalov) [inline doc/tutorials/objdetect/aruco_board_detection/aruco_board_detection.markdown:57]: What does it mean? I propose to replace images, if drown axis are do not correspond to modern OpenCV behaviour.

File: doc/tutorials/objdetect/aruco_board_detection/aruco_board_detection.markdown [hunk 15]
```diff
@@ -0,0 +1,194 @@
...
+So, this board will be composed by 5x7=35 markers. The ids of each of the markers are assigned, by default, in ascending
+order starting on 0, so they will be 0, 1, 2, ..., 34.
+
+After creating a grid board, we probably want to print it and use it. A function to generate the image
+of a `cv::aruco::GridBoard` is provided in `cv::aruco::GridBoard::generateImage()`:
+
+@snippet samples/cpp/tutorial_code/objectDetection/create_board.cpp aruco_generate_board_image
+
+
+A full working example of board creation is included in the `samples/cpp/tutorial_code/objectDetection/create_board.cpp`
```
Comment (asmorkalov) [inline doc/tutorials/objdetect/aruco_board_detection/aruco_board_detection.markdown:108]: I propose to recommend `doc/patter_tools/gen_pattern.py`. As it generates pattern with predictable geometry that is not corrupted with resize or printing scale.

File: doc/tutorials/objdetect/aruco_detection/aruco_detection.markdown [hunk 16]
```diff
@@ -0,0 +1,802 @@
...
+dictionary have the ids: 0, 1, 2, 3 and 4.
+
+More information about dictionaries is provided in the "Selecting a dictionary" section.
+
+
+Marker Creation
+------
+
+Before their detection, markers need to be printed in order to be placed in the environment.
+Marker images can be generated using the `generateImageMarker()` function.
```
Comment (asmorkalov) [inline doc/tutorials/objdetect/aruco_detection/aruco_detection.markdown:66]: Please add gen_pattern.py reference, if it supports dedicated aruco markers.

File: doc/tutorials/objdetect/aruco_detection/aruco_detection.markdown [hunk 17]
```diff
@@ -0,0 +1,802 @@
...
+As a result of the calibration, you get a camera matrix: a matrix of 3x3 elements with the
+focal distances and the camera center coordinates (a.k.a intrinsic parameters), and the distortion
+coefficients: a vector of 5 or more elements that models the distortion produced by your camera.
+
+When you estimate the pose with ArUco markers, you can estimate the pose of each marker individually.
+If you want to estimate one pose from a set of markers, use ArUco Boards (see the **Detection of ArUco
+Boards** tutorial). Using ArUco boards instead of single markers allows some markers to be occluded.
+
+The camera pose relative to the marker is a 3d transformation from the marker coordinate system to the
+camera coordinate system. It is specified by rotation and translation vectors (see `cv::solvePnP()` function for more
```
Comment (asmorkalov) [inline doc/tutorials/objdetect/aruco_detection/aruco_detection.markdown:245]: It is specified by rotation and translation vectors. OpenCV provides cv::solvePnP function to do that.

File: doc/tutorials/objdetect/aruco_detection/aruco_detection.markdown [hunk 18]
```diff
@@ -0,0 +1,802 @@
...
+    cv::drawFrameAxes(outputImage, cameraMatrix, distCoeffs, rvec, tvec, 0.1);
+}
+@endcode
+
+- `outputImage ` is the input/output image where the markers will be drawn (it will normally be the same image where the markers were detected).
+- `cameraMatrix` and `distCoeffs` are the camera calibration parameters.
+- `rvec` and `tvec` are the pose parameters for the marker whose axis is to be drawn.
+- The last parameter is the length of the axis, in the same unit as tvec (usually meters).
+
+A basic full example for pose estimation from single markers:
```
Comment (asmorkalov) [inline doc/tutorials/objdetect/aruco_detection/aruco_detection.markdown:295]: Please rewrite the sentence.

File: doc/tutorials/objdetect/aruco_detection/aruco_detection.markdown [hunk 19]
```diff
@@ -0,0 +1,802 @@
...
+
+@code{.cpp}
```
Comment (asmorkalov) [inline doc/tutorials/objdetect/aruco_detection/aruco_detection.markdown:297]: Please use `@snippet` here.

File: doc/tutorials/objdetect/aruco_detection/aruco_detection.markdown [hunk 20]
```diff
@@ -0,0 +1,802 @@
...
+Note: The samples now take input from the command line using cv::CommandLineParser. For this file the example parameters will look like
+@code{.cpp}
+-v=/path_to_opencv/opencv/doc/tutorials/objdetect/aruco_detection/images/singlemarkersoriginal.jpg -d=10
+-c=/path_to_opencv/opencv/samples/cpp/tutorial_code/objectDetection/tutorial_camera_params.yml
+@endcode
+Parameters for `detect_markers.cpp`:
+@snippet samples/cpp/tutorial_code/objectDetection/detect_markers.cpp aruco_detect_markers_keys
+
+@note To work with examples from the tutorial, you can use camera parameters from `tutorial_camera_params.yml`.
+An example of use in `detect_markers.cpp`.
```
Comment (asmorkalov) [inline doc/tutorials/objdetect/aruco_detection/aruco_detection.markdown:370]: The file is already mentioned in usage example. The not is not needed.

File: doc/tutorials/objdetect/aruco_detection/aruco_detection.markdown [hunk 21]
```diff
@@ -0,0 +1,802 @@
...
+the inter-marker distance. The inter-marker distance is the minimum distance between dictionary markers
+that determines the dictionary's ability to detect and correct errors.
+
+In general, smaller dictionary sizes and larger marker sizes increase the inter-marker distance and
+vice versa. However, the detection of markers with larger sizes is more difficult due to the higher
+number of bits that need to be extracted from the image.
+
+For instance, if you need only 10 markers in your application, it is better to use a dictionary composed only of those 10 markers than using a dictionary composed of 1000 markers. The reason is that
+the dictionary composed of 10 markers will have a higher inter-marker distance and, thus, it will be
+more robust to errors.
```
Comment (asmorkalov) [inline doc/tutorials/objdetect/aruco_detection/aruco_detection.markdown:389]: Formatting.

File: doc/tutorials/objdetect/aruco_detection/aruco_detection.markdown [hunk 22]
```diff
@@ -0,0 +1,802 @@
...
+A dictionary can be generated automatically to adjust the desired number of markers and bits to optimize the inter-marker distance:
+
+@code{.cpp}
+cv::aruco::Dictionary dictionary = cv::aruco::extendDictionary(36, 5);
+@endcode
+
+This will generate a customized dictionary composed of 36 markers of 5x5 bits. The process can take several
+seconds, depending on the parameters (it is slower for larger dictionaries and higher numbers of bits).
+
+### Manual dictionary generation
```
Comment (asmorkalov) [inline doc/tutorials/objdetect/aruco_detection/aruco_detection.markdown:423]: not generation, but definition.

File: doc/tutorials/objdetect/aruco_detection/aruco_detection.markdown [hunk 23]
```diff
@@ -0,0 +1,802 @@
...
+### Automatic dictionary generation
+
+A dictionary can be generated automatically to adjust the desired number of markers and bits to optimize the inter-marker distance:
+
+@code{.cpp}
+cv::aruco::Dictionary dictionary = cv::aruco::extendDictionary(36, 5);
+@endcode
+
+This will generate a customized dictionary composed of 36 markers of 5x5 bits. The process can take several
+seconds, depending on the parameters (it is slower for larger dictionaries and higher numbers of bits).
```
Comment (asmorkalov) [inline doc/tutorials/objdetect/aruco_detection/aruco_detection.markdown:421]: I propose to add reference to dictionary generator tool here.

File: doc/tutorials/objdetect/aruco_detection/aruco_detection.markdown [hunk 24]
```diff
@@ -0,0 +1,802 @@
...
+        cv::Mat bytesList;      // marker code information
+        int markerSize;         // number of bits per dimension
+        int maxCorrectionBits;  // maximum number of bits that can be corrected
+
+        ...
+
+    }
+@endcode
+
+<code>bytesList</code> is the array that contains all the information about the marker codes. `markerSize` is the size
```
Comment (asmorkalov) [inline doc/tutorials/objdetect/aruco_detection/aruco_detection.markdown:444]: Why do you need tags here?

File: doc/tutorials/objdetect/aruco_detection/aruco_detection.markdown [hunk 25]
```diff
@@ -0,0 +1,802 @@
...
+        // markerBits is a 6x6 matrix of CV_8UC1 type, only containing 0s and 1s
+        cv::Mat markerBits = generateMarkerBits();
+        cv::Mat markerCompressed = cv::aruco::Dictionary::getByteListFromBits(markerBits);
+
+        // Add the marker as a new row
+        dictionary.bytesList.push_back(markerCompressed);
+    }
+@endcode
+
+
```
Comment (asmorkalov) [inline doc/tutorials/objdetect/aruco_detection/aruco_detection.markdown:478]: extra space lines.

File: doc/tutorials/objdetect/aruco_detection/aruco_detection.markdown [hunk 26]
```diff
@@ -0,0 +1,802 @@
...
+and performance. All the considered contours will be processed in the following stages, which usually have
+a higher computational cost. So, it is preferred to discard invalid candidates in this stage than in the later stages.
+
+On the other hand, if the filtering conditions are too strict, the real marker contours could be discarded and,
+hence, not detected.
+
+#### minMarkerPerimeterRate and maxMarkerPerimeterRate
+
+These parameters determine the minimum and maximum size of a marker, specifically the minimum and maximum marker perimeter. They are not specified in absolute pixel values, instead they are
+specified relative to the maximum dimension of the input image.
```
Comment (asmorkalov) [inline doc/tutorials/objdetect/aruco_detection/aruco_detection.markdown:557]: formatting.

File: doc/tutorials/objdetect/aruco_detection/aruco_detection.markdown [hunk 27]
```diff
@@ -0,0 +1,802 @@
...
+This penalization is not so noticeable for the `maxMarkerPerimeterRate` parameter, since there are
+usually many more small contours than big contours.
+A `minMarkerPerimeterRate` value of 0 and a `maxMarkerPerimeterRate` value of 4 (or more) will be
+equivalent to consider all the contours in the image, however this is not recommended for
+performance reasons.
+
+Default values:
+
+- `double cv::aruco::DetectorParameters::minMarkerPerimeterRate = 0.03`
+- `double cv::aruco::DetectorParameters::maxMarkerPerimeterRate = 4.0`
```
Comment (asmorkalov) [inline doc/tutorials/objdetect/aruco_detection/aruco_detection.markdown:574]: It's bad idea. Defaults may change in code, but not tutorial. I propose to add reference to API description, but not mention defaults in tutorial text.

File: doc/tutorials/objdetect/aruco_board_detection/aruco_board_detection.markdown [hunk 28]
```diff
@@ -0,0 +1,194 @@
...
+    -w=5 -h=7 -l=100 -s=10
+    -v=/path_to_opencv/opencv/doc/tutorials/objdetect/aruco_board_detection/gboriginal.jpg
+    -c=/path_to_opencv/opencv/samples/cpp/tutorial_code/objectDetection/tutorial_camera_params.yml
+    -cd=/path_to_opencv/opencv/samples/cpp/tutorial_code/objectDetection/tutorial_dict.yml
+@endcode
+Parameters for `detect_board.cpp`:
+@snippet samples/cpp/tutorial_code/objectDetection/detect_board.cpp aruco_detect_board_keys
+@note To work with examples from the tutorial, you can use camera parameters from `tutorial_camera_params.yml` and
+you need use custom dictionary from `tutorial_dict.yml`.
+An example of usage in `detect_board.cpp`.
```
Comment (asmorkalov) [inline doc/tutorials/objdetect/aruco_board_detection/aruco_board_detection.markdown:148]: It's already mentioned in the example. Note is not needed.

File: doc/tutorials/objdetect/aruco_board_detection/aruco_board_detection.markdown (hunk 2 above)
Comment (asmorkalov) [reply doc/tutorials/objdetect/aruco_board_detection/aruco_board_detection.markdown]: The same is true for intermediate images that show algorithm steps.

**Comment by asmorkalov** [comment]
General notes:
- please add standard pre-phrase with text author, compatible OpenCV version, etc.
- not all images are converted to jpeg. I propose to convert intermediate images with contours to jpeg too.
- need to update images with axes  where behaviour is changed. M.b. need to mention the change (opencv version and previous behaviour) as `@note`.

File: doc/tutorials/objdetect/aruco_board_detection/aruco_board_detection.markdown (hunk 14 above)
Comment (AleksandrPanov) [reply doc/tutorials/objdetect/aruco_board_detection/aruco_board_detection.markdown:57]: I don't have the original image to replace the axes. You'll have to use a new image.

Comment (asmorkalov) [reply doc/tutorials/objdetect/aruco_board_detection/aruco_board_detection.markdown:57]: Ok, let's use the new one.

File: doc/tutorials/objdetect/aruco_board_detection/aruco_board_detection.markdown (hunk 2 above)
Comment (AleksandrPanov) [reply doc/tutorials/objdetect/aruco_board_detection/aruco_board_detection.markdown]: The last remaining PNG images are smaller in size than JPG images.

File: doc/tutorials/objdetect/aruco_detection/aruco_detection.markdown (hunk 24 above)
Comment (AleksandrPanov) [reply doc/tutorials/objdetect/aruco_detection/aruco_detection.markdown]: removed tag

File: doc/tutorials/objdetect/aruco_detection/aruco_detection.markdown (hunk 25 above)
Comment (AleksandrPanov) [reply doc/tutorials/objdetect/aruco_detection/aruco_detection.markdown]: fixed

File: doc/tutorials/objdetect/aruco_detection/aruco_detection.markdown (hunk 24 above)
Comment (AleksandrPanov) [reply doc/tutorials/objdetect/aruco_detection/aruco_detection.markdown]: Fixed

File: doc/tutorials/objdetect/aruco_detection/aruco_detection.markdown (hunk 16 above)
Comment (AleksandrPanov) [reply doc/tutorials/objdetect/aruco_detection/aruco_detection.markdown:66]: gen_pattern.py doesn't support the gereneration of single markers

File: doc/tutorials/objdetect/aruco_detection/aruco_detection.markdown (hunk 17 above)
Comment (AleksandrPanov) [reply doc/tutorials/objdetect/aruco_detection/aruco_detection.markdown]: fixed

File: doc/tutorials/objdetect/aruco_detection/aruco_detection.markdown (hunk 18 above)
Comment (AleksandrPanov) [reply doc/tutorials/objdetect/aruco_detection/aruco_detection.markdown]: image

Comment (AleksandrPanov) [reply doc/tutorials/objdetect/aruco_detection/aruco_detection.markdown]: image

File: doc/tutorials/objdetect/aruco_detection/aruco_detection.markdown (hunk 19 above)
Comment (AleksandrPanov) [reply doc/tutorials/objdetect/aruco_detection/aruco_detection.markdown]: Added `@snippet`

File: doc/tutorials/objdetect/aruco_detection/aruco_detection.markdown (hunk 21 above)
Comment (AleksandrPanov) [reply doc/tutorials/objdetect/aruco_detection/aruco_detection.markdown]: Fixed

File: doc/tutorials/objdetect/aruco_detection/aruco_detection.markdown (hunk 22 above)
Comment (AleksandrPanov) [reply doc/tutorials/objdetect/aruco_detection/aruco_detection.markdown]: fixed:

`### Manual dictionary definition`

File: doc/tutorials/objdetect/aruco_detection/aruco_detection.markdown (hunk 26 above)
Comment (AleksandrPanov) [reply doc/tutorials/objdetect/aruco_detection/aruco_detection.markdown]: fixed

File: doc/tutorials/objdetect/aruco_detection/aruco_detection.markdown (hunk 27 above)
Comment (AleksandrPanov) [reply doc/tutorials/objdetect/aruco_detection/aruco_detection.markdown]: fixed, added link in `@see` section:

image

File: doc/tutorials/objdetect/aruco_board_detection/aruco_board_detection.markdown (hunk 28 above)
Comment (AleksandrPanov) [reply doc/tutorials/objdetect/aruco_board_detection/aruco_board_detection.markdown]: deleted

**Comment by AleksandrPanov** [comment]
Added new image for board with occlusions:

gbocclusion

original
original_gbocclusion

File: doc/tutorials/objdetect/aruco_board_detection/aruco_board_detection.markdown [hunk 29]
```diff
@@ -0,0 +1,189 @@
...
+And this is another example with the board partially occluded:
+
+![Board with occlusions](images/gbocclusion.jpg)
+
+As it can be observed, although some markers have not been detected, the board pose can still be estimated from the rest of markers.
+
+Sample video:
+
+@htmlonly
+<iframe width="420" height="315" src="https://www.youtube.com/embed/Q1HlJEjW_j0" frameborder="0" allowfullscreen></iframe>
```
Comment (mshabunin) [inline doc/tutorials/objdetect/aruco_board_detection/aruco_board_detection.markdown:63]: We have doxygen [command](https://github.com/opencv/opencv/blob/d792086ba66a76707e00ee7fffcec50289837fb4/doc/Doxyfile.in#L41) for this: https://github.com/opencv/opencv/blob/d792086ba66a76707e00ee7fffcec50289837fb4/doc/tutorials/app/video_write.markdown?plain=1#L166
Here and in other places.

**Pull request merged**